app/Svg/Parser.hs

```diff
@@ -26,13 +26,18 @@ import Database.DataType
 import Svg.Database (insertGraph, insertElements, deleteGraphs)
 import Config (graphPath, databasePath)
 import qualified Text.Parsec as P
+import Text.Parsec ((<|>))
 import Text.Parsec.String (Parser)
 import Text.Read (readMaybe)
 import Data.Char (isSpace)
 import qualified Data.Text as T
 import Data.Text.IO as T (readFile)
 import Data.List.Split (splitOn)
+import Data.Functor.Identity (Identity)
 
+-- A Parsec parser with user defined state. This can be made stateless by
+-- assigning () to u.
+type StatefulParser u a = P.ParsecT String u Identity a
```

Comment:
I see what you're trying to do here, but there's the `P.Parsec` type synonym already that we should just use. You can turn the `Parser a` uses below into `P.Parsec String u a`.